### PR TITLE
Remove various references to NRefactory that are not used

### PR DESCRIPTION
**Pull request by Therzok**
This has some minor ABI/API breaks.

Ide and CSharpBinding still reference NRefactory.

### CODE REVIEW

**Comment by Therzok** [comment]
cc @sandyarmstrong

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/DiffParser.cs [hunk 1]
```diff
@@ -55,62 +53,6 @@ public class DiffParser : TypeSystemParser
 		public override System.Threading.Tasks.Task<ParsedDocument> Parse (ParseOptions parseOptions, System.Threading.CancellationToken cancellationToken)
 		{
 			ParsedDocument doc = new DefaultParsedDocument (parseOptions.FileName);
-			
-			DefaultUnresolvedTypeDefinition currentFile = null;
```
Comment (Therzok) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/DiffParser.cs]: This code was not used at all.

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp.CodeRefactorings/ConvertToEnum/ConvertToEnumCodeRefactoringProvider.cs [hunk 2]
```diff
@@ -1,504 +0,0 @@
...
-using ICSharpCode.Decompiler.ILAst;
-using MonoDevelop.Core;
-using MonoDevelop.Ide.Editor;
-
-namespace MonoDevelop.CSharp.Refactoring.CodeActions
-{
-	/// <summary>
-	/// Generates an enumeration from const fields
-	/// </summary>
-	class ConvertToEnumAction : MonoDevelop.CodeActions.CodeActionProvider
```
Comment (sandyarmstrong) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.CodeRefactorings/ConvertToEnum/ConvertToEnumCodeRefactoringProvider.cs]: You should also delete the corresponding commented-out section of CSharpBinding.addin.xml that would have instantiated this class.

Comment (Therzok) [reply main/src/addins/CSharpBinding/MonoDevelop.CSharp.CodeRefactorings/ConvertToEnum/ConvertToEnumCodeRefactoringProvider.cs]: Done!

**Comment by Therzok** [comment]
Also bumped CompatVersion.

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/DiffParser.cs (hunk 1 above)
Comment (sandyarmstrong) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/DiffParser.cs]: As in, what, this method never gets called? Do we need the override at all?

**Comment by Therzok** [comment]
@monojenkins rebase

**Comment by Therzok** [comment]
@monojenkins rebase

**Pull request merged**